### DISCUSSION TITLE
GH C# editor and UR robot

### DISCUSSION THREAD

**Discussion by gegardne** [visose/Robots#110] [Q&A] [answered]
We are trying to modify the c# script in the pick and place example.gh file to show block placement using a UR robot, (which does not have PulseDO commands). The script uses the following lines. We tried to replace PulseDO with SetDO, but we get an error: (CS0305): Using the generic type 'System.Linq.Set<TElement>' requires 1 type arguments.  Also tried to use Custom command with a variable declaration, but not sure how to access the variable. Any suggestions would be appreciated. 

 foreach(var cellTarget in program.Targets)
    {
      var target = cellTarget.ProgramTargets[group];
      if (target.Index != 0)
      {
        var pulse = program.Targets[target.Index - 1].ProgramTargets[group].Commands.FirstOrDefault() as Robots.Commands.PulseDO;

        if (pulse != null)
        {
          if (pulse.DO == 1) remaining -= 1;
          if (pulse.DO == 0) placed += 1;
        }
      }
      blocksRemaining.Add(remaining);
      blocksPlaced.Add(placed);
    }
  }

**Comment by visose** [comment, marked as answer]
Use the command type `Robots.Commands.SetDO`, for example `var set = command as Robots.Commands.SetDO;`. The compile error came from resolving `Set` to `System.Linq.Set<TElement>`, which is not the Robots command. `SetDO` exposes `DO` and `Value`, so the PulseDO checks in the sample can be adapted to check `set.DO` and `set.Value` instead.
